src/Orleans.Core.Abstractions/IDs/GrainTypePrefix.cs

```diff
@@ -32,5 +33,45 @@ public static GrainId GetSystemTargetGrainId(GrainType kind, SiloAddress address
         {
             return GrainId.Create(kind, address.ToParsableString());
         }
+
+        public static GrainId GetSystemTargetGrainId(GrainType kind, SiloAddress address, string extraIdentifier)
+        {
+            if (extraIdentifier is string)
+            {
+                return GrainId.Create(kind, address.ToParsableString() + "+" + extraIdentifier);
+            }
+
+            return GetSystemTargetGrainId(kind, address);
+        }
+
+        public static GrainId ReplaceSystemTargetSilo(GrainId grainId, SiloAddress address)
```

Comment:
You're right. In the ClientId/ObserverId PR (not yet open), a shift to a different pattern is made. I was going to wait until a later PR to make the same change for SystemTargets, but perhaps I'll amend this one instead.

Reply:
At first I was confused by this reply, since I didn't remember that one changing `GrainTypePrefix`. Then it occurred to me that you probably specifically meant the approach of strongly typed classes? that would certainly create a logical place for such functionality to live.

Reply:
That's right. I'll update this PR to follow that pattern, too, since it's a little cleaner to get some of this out of the way earlier on.